doc/neps/roadmap.rst

```diff
@@ -68,69 +70,115 @@ documentation on many topics are missing or outdated. See :ref:`NEP44` for
-
-- Easier custom dtypes:
+We aim to continue making it easier to extend NumPy. The primary topic here is to
+improve the dtype system - see for example :ref:`NEP41` and related NEPs linked
+from it. In NumPy 2.0, a new C API for user-defined dtypes was made public. We aim
+to encourage its usage and improve this API further.
 
-  - Simplify and/or wrap the current C-API
-  - More consistent support for dtype metadata
-  - Support for writing a dtype in Python
+Ideas for new dtypes that may be developed outside of the main NumPy repository
+first, and that could potentially be upstreamed into NumPy later, include:
 
-- Allow adding (a) new string dtype(s). This could be encoded strings with
-  fixed-width storage (e.g., ``utf8`` or ``latin1``), and/or a variable length
-  string dtype. The latter could share an implementation with ``dtype=object``,
-  but be explicitly type-checked.
-  One of these should probably be the default for text data. The current
-  string dtype support is neither efficient nor user friendly.
+- A quad-precision (128-bit) dtype
+- A ``bfloat16`` dtype
+- A fixed-width string dtype which supports encodings (e.g., ``utf8`` or
+  ``latin1``)
+- A unit dtype
 
 
 User experience
 ---------------
 
 Type annotations
 ````````````````
-NumPy 1.20 adds type annotations for most NumPy functionality, so users can use
-tools like `mypy`_ to type check their code and IDEs can improve their support
+Type annotations for most NumPy functionality is complete (although some
+submodules like ``numpy.ma`` are missing return types), so users can use tools
+like `mypy`_ to type check their code and IDEs can improve their support
 for NumPy. Improving those type annotations, for example to support annotating
-array shapes and dtypes, is ongoing.
+array shapes (see `gh-16544 <https://github.com/numpy/numpy/issues/16544>`__),
+is ongoing.
 
 Platform support
 ````````````````
 We aim to increase our support for different hardware architectures. This
 includes adding CI coverage when CI services are available, providing wheels on
-PyPI for POWER8/9 (``ppc64le``), providing better build and install
-documentation, and resolving build issues on other platforms like AIX.
+PyPI for platforms that are in high enough demand (e.g., we added ``musllinux``
+ones for NumPy 2.0), and resolving build issues on platforms that we don't test
+in CI (e.g., AIX).
+
+We intend to write a NEP covering the support levels we provide and what is
+required for a platform to move to a higher tier of support, similar to
+`PEP 11 <https://peps.python.org/pep-0011/>`__.
+
+CPython 3.13 will be the first release to offer a free-threaded build (i.e.,
+a CPython build with the GIL disabled). Work is in progress to support this
+well in NumPy. After that is stable and complete, there may be opportunities to
+actually make use of the potential for performance improvements from
+free-threaded CPython, or make it easier to do so for NumPy's users.
+
```

Comment:
Releasing free-threaded wheels is going to significantly increase the number of wheels on PyPI as we drop Python versions that don't support it. I would hope that at some point we will only need to build for a single Python version.

Reply:
Re number of wheels: that should be a temporary situation. In the discussions around PEP 703, this load on package authors was prominently discussed. The end state should be that the free-threaded becomes the default, and then the "with GIL" build fades away.